src/features/class-effect.php

```diff
@@ -16,18 +16,18 @@ final class Effect implements Feature {
 	/**
 	 * The condition to check.
 	 *
-	 * @var callable
+	 * @var bool|callable
 	 */
 	private $when;
 
 	/**
 	 * Constructor.
 	 *
-	 * @param callable $when The condition to check.
-	 * @param Feature  $then The feature to boot if the condition is met.
+	 * @param bool|callable $when The condition to check.
+	 * @param Feature       $then The feature to boot if the condition is met.
 	 */
 	public function __construct(
-		callable $when,
+		bool|callable $when,
```

Comment:
I think this can largely be solved with documentation - we can call out a preference for using the callback for the reasons you've identified, but still have boolean values in case there's a simple comparison that we're doing that can be known ahead of time (e.g., comparing against a constant, like `VIP_GO_APP_ENVIRONMENT` or similar). @srtfisher can you add this detail to the features documentation, please?